src/MICore/Debugger.cs

```diff
@@ -578,20 +578,22 @@ public Task CmdBreak(BreakRequest request)
             return CmdBreakInternal();
         }
 
-        internal bool IsLocalGdb()
+        protected bool IsLocalLaunchUsingServer()
         {
-            return (this.MICommandFactory.Mode == MIMode.Gdb &&
-               this._launchOptions is LocalLaunchOptions &&
-               String.IsNullOrEmpty(((LocalLaunchOptions)this._launchOptions).MIDebuggerServerAddress));
+            return (_launchOptions is LocalLaunchOptions &&
+                (!String.IsNullOrWhiteSpace(((LocalLaunchOptions)_launchOptions).MIDebuggerServerAddress) ||
+                 !String.IsNullOrWhiteSpace(((LocalLaunchOptions)_launchOptions).DebugServer)));
+        }
 
+        internal bool IsLocalGdbTarget()
+        {
+            return (MICommandFactory.Mode == MIMode.Gdb && !IsLocalLaunchUsingServer());
         }
 
-        private bool IsRemoteGdb()
+        private bool IsRemoteGdbTarget()
         {
-            return this.MICommandFactory.Mode == MIMode.Gdb &&
-               (this._launchOptions is PipeLaunchOptions || this._launchOptions is UnixShellPortLaunchOptions ||
-               (this._launchOptions is LocalLaunchOptions
-                    && !String.IsNullOrEmpty(((LocalLaunchOptions)this._launchOptions).MIDebuggerServerAddress)));
+            return MICommandFactory.Mode == MIMode.Gdb &&
+               (_launchOptions is PipeLaunchOptions || _launchOptions is UnixShellPortLaunchOptions || IsLocalLaunchUsingServer());
```

Comment:
nit: if you can add newlines after the `||`, it makes it a little easier to parse

Reply:
Bothered me too I wanted to do it originally but the old line I replaced was about that long.

Do you want me to do it now?

Reply:
If you don't mind

Reply:
Will do. Perhaps I should wait until others comment? Do I close this PR and start a whole new one, or do a separate PR?

Reply:
you can add commits to your branch and push it and it will show up in the PR.

Reply:
Done.